webauthn/helpers/structs.py

```diff
@@ -85,13 +85,11 @@ def _pydantic_v2_validate_bytes_fields(
             """
             field = cls.model_fields[info.field_name]  # type: ignore[attr-defined]
 
-            if field.annotation != bytes:
+            # UserHandle is defined as Optional[bytes]
+            if field.annotation != bytes and info.field_name != 'user_handle':  # type: ignore[attr-defined]
                 return v
 
-            if isinstance(v, str):
-                # NOTE:
-                # Ideally we should only do this when info.mode == "json", but
-                # that does not work when using the deprecated parse_raw method
+            if isinstance(v, str) and info.mode == "json":  # type: ignore[attr-defined]
```

Comment:
Can you please add a comment here explaining what's going on here? It's not obvious to me that `info.mode == "json"` means "when we're converting from JSON to structured data," which is what I'm understanding is happening here after reading your PR description that includes this line:

> The conversion from base64 should only be done if input was json...